clustering.Rmd

```diff
@@ -45,95 +49,147 @@ use the subgroups to generate new questions about the data and follow up with a
+> when onehas an unlabelled data set that is too large to manually label, 
+> but one is willing to provide a few informative example labels as a "seed" 
+> to guess the labels for all the data.
 
 **An illustrative example** 
 
-
 Here we will present an illustrative example using a data set \index{Palmer penguins} from the
 [{palmerpenguins} R data package](https://allisonhorst.github.io/palmerpenguins/). This data set was
 collected by [Dr. Kristen Gorman](https://www.uaf.edu/cfos/people/faculty/detail/kristen-gorman.php) and
-the [Palmer Station, Antarctica LTER](https://lternet.edu/) and includes
-measurements for adult penguins near Palmer Station [@palmerpenguins]. We have
+the [Palmer Station, Antarctica Long Term Ecological Research Site](https://pal.lternet.edu/) and includes
+measurements for adult penguins found near there [@palmerpenguins]. We have
 modified the data set for use in this chapter. Here we will focus on using two
 variables---penguin bill and flipper length, both in millimeters---to determine whether 
 there are distinct types of penguins in our data.
 Understanding this might help us with species discovery and classification in a data-driven
 way.
-```{r 09-penguins, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 3, fig.width = 4, fig.cap = "Gentoo penguin. [@penguinsimage]", fig.retina = 2}
-knitr::include_graphics("img/gentoo.jpg")
+
+```{r 09-penguins, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 3, fig.width = 4, fig.cap = "Gentoo penguin.", fig.retina = 2}
+knitr::include_graphics("img/Gentoo_Penguin._(8671680772).jpg")
 ```
 
-Below we will work with `penguin_data`, a subset of 18 observations of the original data, which has already been scaled. We will discuss scaling for K-means in more detail later in this chapter. \index{mutate}\index{read function!read\_csv} 
+To learn about K-means clustering in this chapter, 
+we will work with `penguin_data`, 
+a subset of 18 observations of the original data, 
+which has already been standardized 
+(remember from Chapter \@ref(classification) that scaling is part of the standardization process). 
+We will discuss scaling for K-means in more detail later in this chapter. 
+\index{mutate}\index{read function!read\_csv} 
+
+Before we get started, we will load the `tidyverse` metapackage to ensure
+we have the functions we need, 
+as well as set a random seed, so that our analysis will be reproducible.
+As we will learn in more detail later in the chapter, 
+the K-means clustering algorithm uses random numbers - 
+which is why setting the seed here is important.
+
 ```{r 10-toy-example-data, echo = FALSE, message = FALSE, warning = FALSE}
 library(tidyverse)
 
 data <- read_csv("data/toy_penguins.csv") |>
   mutate(cluster = as_factor(cluster)) |>
-  mutate(flipper_length_scaled = scale(flipper_length_mm), bill_length_scaled = scale(bill_length_mm))
+  mutate(flipper_length_standardized = as.double(scale(flipper_length_mm)), 
```

Comment:
`as.double` ? looks like a base R function. Is there a `as_double`? if not no big deal, just unused to seeing "dot" functions

Reply:
`as_double` is soft deprecated in favour of `vctrs::vec_cast`. I have moved to that for our own code - but we don't teach that in the book... What do we teach... hmm...

Reply:
This code the students don't see, so I am not going to change it because `vctrs::vec_cast` only works for converting a vector to another vector. What `scale` returns is weird - a matrix. Tidyverse doesn't handle them well, or I don't know the tidyverse version of converting a matrix to a vector (a quick Google search did not help). So I am leaving this as `as.double`. 

Apparently `c` will also convert a matrix to a vector, so could switch to that instead? Still a base R function...